<h1 style="text-align: center;">COMP40610 Information Visualization Project</h1>

### Student Name: Ayush Poojari
### Student Number: 24214170

#### `Importing Libraries`

In [ ]:
import numpy as np
import pandas as pd

#### `Reading original dataset from tidytuesday`

In [ ]:
ol = pd.read_csv('https://raw.githubusercontent.com/rfordatascience/tidytuesday/master/data/2021/2021-07-27/olympics.csv')

#### `Information about dataset`

In [ ]:
ol.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 271116 entries, 0 to 271115
Data columns (total 15 columns):
 #   Column  Non-Null Count   Dtype  
---  ------  --------------   -----  
 0   id      271116 non-null  int64  
 1   name    271116 non-null  object 
 2   sex     271116 non-null  object 
 3   age     261642 non-null  float64
 4   height  210945 non-null  float64
 5   weight  208241 non-null  float64
 6   team    271116 non-null  object 
 7   noc     271116 non-null  object 
 8   games   271116 non-null  object 
 9   year    271116 non-null  int64  
 10  season  271116 non-null  object 
 11  city    271116 non-null  object 
 12  sport   271116 non-null  object 
 13  event   271116 non-null  object 
 14  medal   39783 non-null   object 
dtypes: float64(3), int64(2), object(10)
memory usage: 31.0+ MB


`Insights: 15 features/columns in the dataset`
- `5 Quantitative features/columns: id,age,height,weight and year`
- `10 Categorical features/columns: name,sex,team,noc,games,season,city,sport,event and medal`
- `Memory Usage: 31.0+ MB`

`📌 NOTE: Year feature/column needs to be converted datetime if needed`

#### `Describing the dataset`

In [ ]:
ol.describe().T

count          mean           std     min      25%      50%  \
id      271116.0  68248.954396  39022.286345     1.0  34643.0  68205.0   
age     261642.0     25.556898      6.393561    10.0     21.0     24.0   
height  210945.0    175.338970     10.518462   127.0    168.0    175.0   
weight  208241.0     70.702393     14.348020    25.0     60.0     70.0   
year    271116.0   1978.378480     29.877632  1896.0   1960.0   1988.0   

              75%       max  
id      102097.25  135571.0  
age         28.00      97.0  
height     183.00     226.0  
weight      79.00     214.0  
year      2002.00    2016.0

`📌 Insights: We can get overall picture about quantitative data in the dataset`
- `id: Starts from 1 upto 135571`
- `age: Average age 28.0 years, minimum 10.0 years and maximum 97.0 years`
- `height: Average height 183.0 cm, minimum 127.0 cm and maximum 226.0 cm`
- `weight: Average weight 79.0 kg, minimum 24.0 kg and maximum 214.0 kg`

#### `Viewing the dataset:`

In [ ]:
ol.head(14)

id                      name sex   age  height  weight            team  \
0    1                 A Dijiang   M  24.0   180.0    80.0           China   
1    2                  A Lamusi   M  23.0   170.0    60.0           China   
2    3       Gunnar Nielsen Aaby   M  24.0     NaN     NaN         Denmark   
3    4      Edgar Lindenau Aabye   M  34.0     NaN     NaN  Denmark/Sweden   
4    5  Christine Jacoba Aaftink   F  21.0   185.0    82.0     Netherlands   
5    5  Christine Jacoba Aaftink   F  21.0   185.0    82.0     Netherlands   
6    5  Christine Jacoba Aaftink   F  25.0   185.0    82.0     Netherlands   
7    5  Christine Jacoba Aaftink   F  25.0   185.0    82.0     Netherlands   
8    5  Christine Jacoba Aaftink   F  27.0   185.0    82.0     Netherlands   
9    5  Christine Jacoba Aaftink   F  27.0   185.0    82.0     Netherlands   
10   6           Per Knut Aaland   M  31.0   188.0    75.0   United States   
11   6           Per Knut Aaland   M  31.0   188.0    75.0   United States   
12   6           Per Knut Aaland   M  31.0   188.0    75.0   United States   
13   6           Per Knut Aaland   M  31.0   188.0    75.0   United States   

    noc        games  year  season         city                 sport  \
0   CHN  1992 Summer  1992  Summer    Barcelona            Basketball   
1   CHN  2012 Summer  2012  Summer       London                  Judo   
2   DEN  1920 Summer  1920  Summer    Antwerpen              Football   
3   DEN  1900 Summer  1900  Summer        Paris            Tug-Of-War   
4   NED  1988 Winter  1988  Winter      Calgary         Speed Skating   
5   NED  1988 Winter  1988  Winter      Calgary         Speed Skating   
6   NED  1992 Winter  1992  Winter  Albertville         Speed Skating   
7   NED  1992 Winter  1992  Winter  Albertville         Speed Skating   
8   NED  1994 Winter  1994  Winter  Lillehammer         Speed Skating   
9   NED  1994 Winter  1994  Winter  Lillehammer         Speed Skating   
10  USA  1992 Winter  1992  Winter  Albertville  Cross Country Skiing   
11  USA  1992 Winter  1992  Winter  Albertville  Cross Country Skiing   
12  USA  1992 Winter  1992  Winter  Albertville  Cross Country Skiing   
13  USA  1992 Winter  1992  Winter  Albertville  Cross Country Skiing   

                                                event medal  
0                         Basketball Men's Basketball   NaN  
1                        Judo Men's Extra-Lightweight   NaN  
2                             Football Men's Football   NaN  
3                         Tug-Of-War Men's Tug-Of-War  Gold  
4                    Speed Skating Women's 500 metres   NaN  
5                  Speed Skating Women's 1,000 metres   NaN  
6                    Speed Skating Women's 500 metres   NaN  
7                  Speed Skating Women's 1,000 metres   NaN  
8                    Speed Skating Women's 500 metres   NaN  
9                  Speed Skating Women's 1,000 metres   NaN  
10           Cross Country Skiing Men's 10 kilometres   NaN  
11           Cross Country Skiing Men's 50 kilometres   NaN  
12  Cross Country Skiing Men's 10/15 kilometres Pu...   NaN  
13  Cross Country Skiing Men's 4 x 10 kilometres R...   NaN

In [ ]:
ol.tail(14)

id                          name sex   age  height  weight  \
271102  135563     Olesya Nikolayevna Zykina   F  19.0   171.0    64.0   
271103  135563     Olesya Nikolayevna Zykina   F  23.0   171.0    64.0   
271104  135564  Yevgeny Aleksandrovich Zykov   M  22.0   172.0    65.0   
271105  135565      Fernando scar Zylberberg   M  23.0   168.0    76.0   
271106  135565      Fernando scar Zylberberg   M  27.0   168.0    76.0   
271107  135566    James Francis "Jim" Zylker   M  21.0   175.0    75.0   
271108  135567  Aleksandr Viktorovich Zyuzin   M  24.0   183.0    72.0   
271109  135567  Aleksandr Viktorovich Zyuzin   M  28.0   183.0    72.0   
271110  135568        Olga Igorevna Zyuzkova   F  33.0   171.0    69.0   
271111  135569                    Andrzej ya   M  29.0   179.0    89.0   
271112  135570                      Piotr ya   M  27.0   176.0    59.0   
271113  135570                      Piotr ya   M  27.0   176.0    59.0   
271114  135571            Tomasz Ireneusz ya   M  30.0   185.0    96.0   
271115  135571            Tomasz Ireneusz ya   M  34.0   185.0    96.0   

                 team  noc        games  year  season            city  \
271102         Russia  RUS  2000 Summer  2000  Summer          Sydney   
271103         Russia  RUS  2004 Summer  2004  Summer          Athina   
271104       Russia-1  RUS  2002 Winter  2002  Winter  Salt Lake City   
271105      Argentina  ARG  2000 Summer  2000  Summer          Sydney   
271106      Argentina  ARG  2004 Summer  2004  Summer          Athina   
271107  United States  USA  1972 Summer  1972  Summer          Munich   
271108         Russia  RUS  2000 Summer  2000  Summer          Sydney   
271109         Russia  RUS  2004 Summer  2004  Summer          Athina   
271110        Belarus  BLR  2016 Summer  2016  Summer  Rio de Janeiro   
271111       Poland-1  POL  1976 Winter  1976  Winter       Innsbruck   
271112         Poland  POL  2014 Winter  2014  Winter           Sochi   
271113         Poland  POL  2014 Winter  2014  Winter           Sochi   
271114         Poland  POL  1998 Winter  1998  Winter          Nagano   
271115         Poland  POL  2002 Winter  2002  Winter  Salt Lake City   

              sport                                     event   medal  
271102    Athletics    Athletics Women's 4 x 400 metres Relay  Bronze  
271103    Athletics    Athletics Women's 4 x 400 metres Relay  Silver  
271104         Luge                Luge Mixed (Men)'s Doubles     NaN  
271105       Hockey                       Hockey Men's Hockey     NaN  
271106       Hockey                       Hockey Men's Hockey     NaN  
271107     Football                   Football Men's Football     NaN  
271108       Rowing    Rowing Men's Lightweight Coxless Fours     NaN  
271109       Rowing    Rowing Men's Lightweight Coxless Fours     NaN  
271110   Basketball             Basketball Women's Basketball     NaN  
271111         Luge                Luge Mixed (Men)'s Doubles     NaN  
271112  Ski Jumping  Ski Jumping Men's Large Hill, Individual     NaN  
271113  Ski Jumping        Ski Jumping Men's Large Hill, Team     NaN  
271114    Bobsleigh                      Bobsleigh Men's Four     NaN  
271115    Bobsleigh                      Bobsleigh Men's Four     NaN

#### `Dimension of dataset:`

In [ ]:
print(f'The dataset has {ol.shape[0]} rows/tuples and {ol.shape[1]} features/columns')

The dataset has 271116 rows/tuples and 15 features/columns


In [ ]:
start_year = ol['year'].min()
end_year = ol['year'].max()
print(f"The dataset covers years from {start_year} to {end_year}.")

The dataset covers years from 1896 to 2016.


### `📌 Data Cleaning and Preprocessing`

#### `Checking for missing values in rows/tuples`

In [ ]:
ol.isna().sum()

id             0
name           0
sex            0
age         9474
height     60171
weight     62875
team           0
noc            0
games          0
year           0
season         0
city           0
sport          0
event          0
medal     231333
dtype: int64

In [ ]:
ol.isnull().sum()

id             0
name           0
sex            0
age         9474
height     60171
weight     62875
team           0
noc            0
games          0
year           0
season         0
city           0
sport          0
event          0
medal     231333
dtype: int64

#### `There is following number of missing values:`
- age	9474
- height	60171
- weight	62875
- medal	231333

#### `📌 NOTE: Medals has maximum NA because it represents no medals we can set it "No Medal"`

#### `Dropping missing values from age, height and weight features/columns`

In [ ]:
ol = ol.dropna(subset=['age', 'height', 'weight'], how='any')

In [ ]:
ol.shape

(206165, 15)

In [ ]:
ol.isna().sum()

id             0
name           0
sex            0
age            0
height         0
weight         0
team           0
noc            0
games          0
year           0
season         0
city           0
sport          0
event          0
medal     175984
dtype: int64

In [ ]:
ol.isnull().sum()

id             0
name           0
sex            0
age            0
height         0
weight         0
team           0
noc            0
games          0
year           0
season         0
city           0
sport          0
event          0
medal     175984
dtype: int64

`We can see there are no missing values in the age, height and weight features/columns`

#### `Checking for duplicate rows/tuples in dataset`

In [ ]:
ol[ol.duplicated()].head(13)

id                        name sex   age  height  weight  \
87975    44600                Gavin Hadden   M  44.0   188.0    77.0   
87976    44600                Gavin Hadden   M  44.0   188.0    77.0   
87977    44600                Gavin Hadden   M  44.0   188.0    77.0   
92832    47034        Louis Hechenbleikner   M  38.0   178.0    67.0   
92833    47034        Louis Hechenbleikner   M  38.0   178.0    67.0   
92834    47034        Louis Hechenbleikner   M  38.0   178.0    67.0   
92835    47034        Louis Hechenbleikner   M  38.0   178.0    67.0   
151807   76185           Alexandre Maspoli   M  48.0   170.0    81.0   
212857  106878     Maurice Auguste Schills   M  20.0   188.0    83.0   
216070  108451        Edgar Isaac Seligman   M  61.0   174.0    80.0   
222410  111729  Jerzy Walerian Skolimowski   M  24.0   150.0    59.0   
222411  111729  Jerzy Walerian Skolimowski   M  24.0   150.0    59.0   
222412  111729  Jerzy Walerian Skolimowski   M  24.0   150.0    59.0   

                 team  noc        games  year  season         city  \
87975   United States  USA  1932 Summer  1932  Summer  Los Angeles   
87976   United States  USA  1932 Summer  1932  Summer  Los Angeles   
87977   United States  USA  1932 Summer  1932  Summer  Los Angeles   
92832   United States  USA  1932 Summer  1932  Summer  Los Angeles   
92833   United States  USA  1932 Summer  1932  Summer  Los Angeles   
92834   United States  USA  1932 Summer  1932  Summer  Los Angeles   
92835   United States  USA  1932 Summer  1932  Summer  Los Angeles   
151807         France  FRA  1924 Summer  1924  Summer        Paris   
212857         France  FRA  1908 Summer  1908  Summer       London   
216070  Great Britain  GBR  1928 Summer  1928  Summer    Amsterdam   
222410         Poland  POL  1932 Summer  1932  Summer  Los Angeles   
222411         Poland  POL  1932 Summer  1932  Summer  Los Angeles   
222412         Poland  POL  1932 Summer  1932  Summer  Los Angeles   

                   sport                                              event  \
87975   Art Competitions  Art Competitions Mixed Architecture, Unknown E...   
87976   Art Competitions  Art Competitions Mixed Architecture, Unknown E...   
87977   Art Competitions  Art Competitions Mixed Architecture, Unknown E...   
92832   Art Competitions     Art Competitions Mixed Painting, Unknown Event   
92833   Art Competitions     Art Competitions Mixed Painting, Unknown Event   
92834   Art Competitions     Art Competitions Mixed Painting, Unknown Event   
92835   Art Competitions     Art Competitions Mixed Painting, Unknown Event   
151807  Art Competitions                 Art Competitions Mixed Sculpturing   
212857           Cycling                               Cycling Men's Sprint   
216070  Art Competitions         Art Competitions Mixed Painting, Paintings   
222410  Art Competitions     Art Competitions Mixed Painting, Unknown Event   
222411  Art Competitions     Art Competitions Mixed Painting, Unknown Event   
222412  Art Competitions     Art Competitions Mixed Painting, Unknown Event   

       medal  
87975    NaN  
87976    NaN  
87977    NaN  
92832    NaN  
92833    NaN  
92834    NaN  
92835    NaN  
151807   NaN  
212857   NaN  
216070   NaN  
222410   NaN  
222411   NaN  
222412   NaN

In [ ]:
ol[ol.duplicated()].shape

(13, 15)

In [ ]:
ol = ol.drop_duplicates()

In [ ]:
ol.shape

(206152, 15)

`Dropped all duplicate rows and tuples`

### `📌 In Medal Glory: Olympic Medallists and Winning Nations (1896–2016) dashboard 📊`

> `I am visualizing metrics and statistics only for olympic medallist and top winning nations`

> `The required step do so is to filters out only medal winning players`

`Dropped columns where medal is NA which means "No medal"`


In [ ]:
ol.dropna(subset=['medal'],inplace=True)

In [ ]:
ol.shape

(30181, 15)

#### `Dimension of dataset is reduced to (30181,15)`

In [ ]:
ol.head(12)

id                      name sex   age  height  weight     team  noc  \
40  16  Juhamatti Tapio Aaltonen   M  28.0   184.0    85.0  Finland  FIN   
41  17   Paavo Johannes Aaltonen   M  28.0   175.0    64.0  Finland  FIN   
42  17   Paavo Johannes Aaltonen   M  28.0   175.0    64.0  Finland  FIN   
44  17   Paavo Johannes Aaltonen   M  28.0   175.0    64.0  Finland  FIN   
48  17   Paavo Johannes Aaltonen   M  28.0   175.0    64.0  Finland  FIN   
50  17   Paavo Johannes Aaltonen   M  32.0   175.0    64.0  Finland  FIN   
60  20        Kjetil Andr Aamodt   M  20.0   176.0    85.0   Norway  NOR   
61  20        Kjetil Andr Aamodt   M  20.0   176.0    85.0   Norway  NOR   
63  20        Kjetil Andr Aamodt   M  22.0   176.0    85.0   Norway  NOR   
64  20        Kjetil Andr Aamodt   M  22.0   176.0    85.0   Norway  NOR   
67  20        Kjetil Andr Aamodt   M  22.0   176.0    85.0   Norway  NOR   
73  20        Kjetil Andr Aamodt   M  30.0   176.0    85.0   Norway  NOR   

          games  year  season            city          sport  \
40  2014 Winter  2014  Winter           Sochi     Ice Hockey   
41  1948 Summer  1948  Summer          London     Gymnastics   
42  1948 Summer  1948  Summer          London     Gymnastics   
44  1948 Summer  1948  Summer          London     Gymnastics   
48  1948 Summer  1948  Summer          London     Gymnastics   
50  1952 Summer  1952  Summer        Helsinki     Gymnastics   
60  1992 Winter  1992  Winter     Albertville  Alpine Skiing   
61  1992 Winter  1992  Winter     Albertville  Alpine Skiing   
63  1994 Winter  1994  Winter     Lillehammer  Alpine Skiing   
64  1994 Winter  1994  Winter     Lillehammer  Alpine Skiing   
67  1994 Winter  1994  Winter     Lillehammer  Alpine Skiing   
73  2002 Winter  2002  Winter  Salt Lake City  Alpine Skiing   

                                     event   medal  
40             Ice Hockey Men's Ice Hockey  Bronze  
41  Gymnastics Men's Individual All-Around  Bronze  
42        Gymnastics Men's Team All-Around    Gold  
44            Gymnastics Men's Horse Vault    Gold  
48        Gymnastics Men's Pommelled Horse    Gold  
50        Gymnastics Men's Team All-Around  Bronze  
60             Alpine Skiing Men's Super G    Gold  
61        Alpine Skiing Men's Giant Slalom  Bronze  
63            Alpine Skiing Men's Downhill  Silver  
64             Alpine Skiing Men's Super G  Bronze  
67            Alpine Skiing Men's Combined  Silver  
73             Alpine Skiing Men's Super G    Gold

#### `Making dataset consistent`

In [ ]:
ol.columns = ol.columns.str.lower().str.replace(' ', '_')

In [ ]:
ol.columns

Index(['id', 'name', 'sex', 'age', 'height', 'weight', 'team', 'noc', 'games',
       'year', 'season', 'city', 'sport', 'event', 'medal'],
      dtype='object')

#### `Checking for outliers or erroneous values`

In [ ]:
ol.describe().T

count          mean           std     min      25%      50%  \
id      30181.0  70225.949604  38839.720551    16.0  37494.0  69771.0   
age     30181.0     25.429012      5.049684    13.0     22.0     25.0   
height  30181.0    177.642358     10.924188   136.0    170.0    178.0   
weight  30181.0     73.753554     15.004992    28.0     63.0     73.0   
year    30181.0   1988.005964     22.718451  1896.0   1976.0   1992.0   

             75%       max  
id      104111.0  135563.0  
age         28.0      66.0  
height     185.0     223.0  
weight      83.0     182.0  
year      2006.0    2016.0

In [ ]:
ol = ol[(ol['age'] >= 10) & (ol['age'] < 120)]

In [ ]:
ol = ol[(ol['height'] < 300)]

In [ ]:
ol.shape

(30181, 15)

In [ ]:
ol.head(14)

id                      name sex   age  height  weight     team  noc  \
40  16  Juhamatti Tapio Aaltonen   M  28.0   184.0    85.0  Finland  FIN   
41  17   Paavo Johannes Aaltonen   M  28.0   175.0    64.0  Finland  FIN   
42  17   Paavo Johannes Aaltonen   M  28.0   175.0    64.0  Finland  FIN   
44  17   Paavo Johannes Aaltonen   M  28.0   175.0    64.0  Finland  FIN   
48  17   Paavo Johannes Aaltonen   M  28.0   175.0    64.0  Finland  FIN   
50  17   Paavo Johannes Aaltonen   M  32.0   175.0    64.0  Finland  FIN   
60  20        Kjetil Andr Aamodt   M  20.0   176.0    85.0   Norway  NOR   
61  20        Kjetil Andr Aamodt   M  20.0   176.0    85.0   Norway  NOR   
63  20        Kjetil Andr Aamodt   M  22.0   176.0    85.0   Norway  NOR   
64  20        Kjetil Andr Aamodt   M  22.0   176.0    85.0   Norway  NOR   
67  20        Kjetil Andr Aamodt   M  22.0   176.0    85.0   Norway  NOR   
73  20        Kjetil Andr Aamodt   M  30.0   176.0    85.0   Norway  NOR   
76  20        Kjetil Andr Aamodt   M  30.0   176.0    85.0   Norway  NOR   
78  20        Kjetil Andr Aamodt   M  34.0   176.0    85.0   Norway  NOR   

          games  year  season            city          sport  \
40  2014 Winter  2014  Winter           Sochi     Ice Hockey   
41  1948 Summer  1948  Summer          London     Gymnastics   
42  1948 Summer  1948  Summer          London     Gymnastics   
44  1948 Summer  1948  Summer          London     Gymnastics   
48  1948 Summer  1948  Summer          London     Gymnastics   
50  1952 Summer  1952  Summer        Helsinki     Gymnastics   
60  1992 Winter  1992  Winter     Albertville  Alpine Skiing   
61  1992 Winter  1992  Winter     Albertville  Alpine Skiing   
63  1994 Winter  1994  Winter     Lillehammer  Alpine Skiing   
64  1994 Winter  1994  Winter     Lillehammer  Alpine Skiing   
67  1994 Winter  1994  Winter     Lillehammer  Alpine Skiing   
73  2002 Winter  2002  Winter  Salt Lake City  Alpine Skiing   
76  2002 Winter  2002  Winter  Salt Lake City  Alpine Skiing   
78  2006 Winter  2006  Winter          Torino  Alpine Skiing   

                                     event   medal  
40             Ice Hockey Men's Ice Hockey  Bronze  
41  Gymnastics Men's Individual All-Around  Bronze  
42        Gymnastics Men's Team All-Around    Gold  
44            Gymnastics Men's Horse Vault    Gold  
48        Gymnastics Men's Pommelled Horse    Gold  
50        Gymnastics Men's Team All-Around  Bronze  
60             Alpine Skiing Men's Super G    Gold  
61        Alpine Skiing Men's Giant Slalom  Bronze  
63            Alpine Skiing Men's Downhill  Silver  
64             Alpine Skiing Men's Super G  Bronze  
67            Alpine Skiing Men's Combined  Silver  
73             Alpine Skiing Men's Super G    Gold  
76            Alpine Skiing Men's Combined    Gold  
78             Alpine Skiing Men's Super G    Gold

In [ ]:
ol.tail(14)

id                                 name sex   age  height  weight  \
271013  135522      Klaas Erik "Klaas-Erik" Zwering   M  23.0   189.0    80.0   
271015  135523       Ronald Ferdinand "Ron" Zwerver   M  25.0   200.0    93.0   
271016  135523       Ronald Ferdinand "Ron" Zwerver   M  29.0   200.0    93.0   
271019  135525                       Martin Zwicker   M  29.0   175.0    64.0   
271032  135535            Claudia Antoinette Zwiers   F  22.0   181.0    78.0   
271046  135544                   Krzysztof Zwoliski   M  21.0   175.0    70.0   
271048  135545                      Henk Jan Zwolle   M  27.0   197.0    93.0   
271049  135545                      Henk Jan Zwolle   M  31.0   197.0    93.0   
271076  135553  Galina Ivanovna Zybina (-Fyodorova)   F  21.0   168.0    80.0   
271078  135553  Galina Ivanovna Zybina (-Fyodorova)   F  25.0   168.0    80.0   
271080  135553  Galina Ivanovna Zybina (-Fyodorova)   F  33.0   168.0    80.0   
271082  135554                         Bogusaw Zych   M  28.0   182.0    82.0   
271102  135563            Olesya Nikolayevna Zykina   F  19.0   171.0    64.0   
271103  135563            Olesya Nikolayevna Zykina   F  23.0   171.0    64.0   

                team  noc        games  year  season            city  \
271013   Netherlands  NED  2004 Summer  2004  Summer          Athina   
271015   Netherlands  NED  1992 Summer  1992  Summer       Barcelona   
271016   Netherlands  NED  1996 Summer  1996  Summer         Atlanta   
271019       Germany  GER  2016 Summer  2016  Summer  Rio de Janeiro   
271032   Netherlands  NED  1996 Summer  1996  Summer         Atlanta   
271046        Poland  POL  1980 Summer  1980  Summer          Moskva   
271048   Netherlands  NED  1992 Summer  1992  Summer       Barcelona   
271049   Netherlands  NED  1996 Summer  1996  Summer         Atlanta   
271076  Soviet Union  URS  1952 Summer  1952  Summer        Helsinki   
271078  Soviet Union  URS  1956 Summer  1956  Summer       Melbourne   
271080  Soviet Union  URS  1964 Summer  1964  Summer           Tokyo   
271082        Poland  POL  1980 Summer  1980  Summer          Moskva   
271102        Russia  RUS  2000 Summer  2000  Summer          Sydney   
271103        Russia  RUS  2004 Summer  2004  Summer          Athina   

             sport                                          event   medal  
271013    Swimming  Swimming Men's 4 x 100 metres Freestyle Relay  Silver  
271015  Volleyball                    Volleyball Men's Volleyball  Silver  
271016  Volleyball                    Volleyball Men's Volleyball    Gold  
271019      Hockey                            Hockey Men's Hockey  Bronze  
271032        Judo                      Judo Women's Middleweight  Bronze  
271046   Athletics           Athletics Men's 4 x 100 metres Relay  Silver  
271048      Rowing                     Rowing Men's Double Sculls  Bronze  
271049      Rowing                      Rowing Men's Coxed Eights    Gold  
271076   Athletics                     Athletics Women's Shot Put    Gold  
271078   Athletics                     Athletics Women's Shot Put  Silver  
271080   Athletics                     Athletics Women's Shot Put  Bronze  
271082     Fencing                       Fencing Men's Foil, Team  Bronze  
271102   Athletics         Athletics Women's 4 x 400 metres Relay  Bronze  
271103   Athletics         Athletics Women's 4 x 400 metres Relay  Silver

#### `📌 Feature Transformation: Deriving features/columns`

#### `BMI from height and weight`

In [ ]:
ol['bmi'] = ol['weight'] / (ol['height'] / 100) ** 2

In [ ]:
ol.head(8)

id                      name sex   age  height  weight     team  noc  \
40  16  Juhamatti Tapio Aaltonen   M  28.0   184.0    85.0  Finland  FIN   
41  17   Paavo Johannes Aaltonen   M  28.0   175.0    64.0  Finland  FIN   
42  17   Paavo Johannes Aaltonen   M  28.0   175.0    64.0  Finland  FIN   
44  17   Paavo Johannes Aaltonen   M  28.0   175.0    64.0  Finland  FIN   
48  17   Paavo Johannes Aaltonen   M  28.0   175.0    64.0  Finland  FIN   
50  17   Paavo Johannes Aaltonen   M  32.0   175.0    64.0  Finland  FIN   
60  20        Kjetil Andr Aamodt   M  20.0   176.0    85.0   Norway  NOR   
61  20        Kjetil Andr Aamodt   M  20.0   176.0    85.0   Norway  NOR   

          games  year  season         city          sport  \
40  2014 Winter  2014  Winter        Sochi     Ice Hockey   
41  1948 Summer  1948  Summer       London     Gymnastics   
42  1948 Summer  1948  Summer       London     Gymnastics   
44  1948 Summer  1948  Summer       London     Gymnastics   
48  1948 Summer  1948  Summer       London     Gymnastics   
50  1952 Summer  1952  Summer     Helsinki     Gymnastics   
60  1992 Winter  1992  Winter  Albertville  Alpine Skiing   
61  1992 Winter  1992  Winter  Albertville  Alpine Skiing   

                                     event   medal        bmi  
40             Ice Hockey Men's Ice Hockey  Bronze  25.106333  
41  Gymnastics Men's Individual All-Around  Bronze  20.897959  
42        Gymnastics Men's Team All-Around    Gold  20.897959  
44            Gymnastics Men's Horse Vault    Gold  20.897959  
48        Gymnastics Men's Pommelled Horse    Gold  20.897959  
50        Gymnastics Men's Team All-Around  Bronze  20.897959  
60             Alpine Skiing Men's Super G    Gold  27.440599  
61        Alpine Skiing Men's Giant Slalom  Bronze  27.440599

In [ ]:
ol['ratio'] = ol['height'] / ol['weight']

In [ ]:
ol.head(8)

id                      name sex   age  height  weight     team  noc  \
40  16  Juhamatti Tapio Aaltonen   M  28.0   184.0    85.0  Finland  FIN   
41  17   Paavo Johannes Aaltonen   M  28.0   175.0    64.0  Finland  FIN   
42  17   Paavo Johannes Aaltonen   M  28.0   175.0    64.0  Finland  FIN   
44  17   Paavo Johannes Aaltonen   M  28.0   175.0    64.0  Finland  FIN   
48  17   Paavo Johannes Aaltonen   M  28.0   175.0    64.0  Finland  FIN   
50  17   Paavo Johannes Aaltonen   M  32.0   175.0    64.0  Finland  FIN   
60  20        Kjetil Andr Aamodt   M  20.0   176.0    85.0   Norway  NOR   
61  20        Kjetil Andr Aamodt   M  20.0   176.0    85.0   Norway  NOR   

          games  year  season         city          sport  \
40  2014 Winter  2014  Winter        Sochi     Ice Hockey   
41  1948 Summer  1948  Summer       London     Gymnastics   
42  1948 Summer  1948  Summer       London     Gymnastics   
44  1948 Summer  1948  Summer       London     Gymnastics   
48  1948 Summer  1948  Summer       London     Gymnastics   
50  1952 Summer  1952  Summer     Helsinki     Gymnastics   
60  1992 Winter  1992  Winter  Albertville  Alpine Skiing   
61  1992 Winter  1992  Winter  Albertville  Alpine Skiing   

                                     event   medal        bmi     ratio  
40             Ice Hockey Men's Ice Hockey  Bronze  25.106333  2.164706  
41  Gymnastics Men's Individual All-Around  Bronze  20.897959  2.734375  
42        Gymnastics Men's Team All-Around    Gold  20.897959  2.734375  
44            Gymnastics Men's Horse Vault    Gold  20.897959  2.734375  
48        Gymnastics Men's Pommelled Horse    Gold  20.897959  2.734375  
50        Gymnastics Men's Team All-Around  Bronze  20.897959  2.734375  
60             Alpine Skiing Men's Super G    Gold  27.440599  2.070588  
61        Alpine Skiing Men's Giant Slalom  Bronze  27.440599  2.070588

In [ ]:
def bmi_category(bmi):
    if bmi < 18.5:
        return 'Underweight'
    elif 18.5 <= bmi < 24.9:
        return 'Normal weight'
    elif 25 <= bmi < 29.9:
        return 'Overweight'
    else:
        return 'Obese'

ol['bmi_category'] = ol['bmi'].apply(bmi_category)

In [ ]:
ol.head(8)

id                      name sex   age  height  weight     team  noc  \
40  16  Juhamatti Tapio Aaltonen   M  28.0   184.0    85.0  Finland  FIN   
41  17   Paavo Johannes Aaltonen   M  28.0   175.0    64.0  Finland  FIN   
42  17   Paavo Johannes Aaltonen   M  28.0   175.0    64.0  Finland  FIN   
44  17   Paavo Johannes Aaltonen   M  28.0   175.0    64.0  Finland  FIN   
48  17   Paavo Johannes Aaltonen   M  28.0   175.0    64.0  Finland  FIN   
50  17   Paavo Johannes Aaltonen   M  32.0   175.0    64.0  Finland  FIN   
60  20        Kjetil Andr Aamodt   M  20.0   176.0    85.0   Norway  NOR   
61  20        Kjetil Andr Aamodt   M  20.0   176.0    85.0   Norway  NOR   

          games  year  season         city          sport  \
40  2014 Winter  2014  Winter        Sochi     Ice Hockey   
41  1948 Summer  1948  Summer       London     Gymnastics   
42  1948 Summer  1948  Summer       London     Gymnastics   
44  1948 Summer  1948  Summer       London     Gymnastics   
48  1948 Summer  1948  Summer       London     Gymnastics   
50  1952 Summer  1952  Summer     Helsinki     Gymnastics   
60  1992 Winter  1992  Winter  Albertville  Alpine Skiing   
61  1992 Winter  1992  Winter  Albertville  Alpine Skiing   

                                     event   medal        bmi     ratio  \
40             Ice Hockey Men's Ice Hockey  Bronze  25.106333  2.164706   
41  Gymnastics Men's Individual All-Around  Bronze  20.897959  2.734375   
42        Gymnastics Men's Team All-Around    Gold  20.897959  2.734375   
44            Gymnastics Men's Horse Vault    Gold  20.897959  2.734375   
48        Gymnastics Men's Pommelled Horse    Gold  20.897959  2.734375   
50        Gymnastics Men's Team All-Around  Bronze  20.897959  2.734375   
60             Alpine Skiing Men's Super G    Gold  27.440599  2.070588   
61        Alpine Skiing Men's Giant Slalom  Bronze  27.440599  2.070588   

     bmi_category  
40     Overweight  
41  Normal weight  
42  Normal weight  
44  Normal weight  
48  Normal weight  
50  Normal weight  
60     Overweight  
61     Overweight

####  `Derived features/columns: bmi, bmi_category and ratio`

In [ ]:
ol['sex'] = ol['sex'].str.capitalize()
ol['team'] = ol['team'].str.capitalize()
ol['season'] = ol['season'].str.capitalize()
ol['city'] = ol['city'].str.capitalize()
ol['sport'] = ol['sport'].str.capitalize()
ol['medal'] = ol['medal'].str.capitalize()
ol['bmi_category'] = ol['bmi_category'].str.capitalize()

In [ ]:
ol.head(4)

id                      name sex   age  height  weight     team  noc  \
40  16  Juhamatti Tapio Aaltonen   M  28.0   184.0    85.0  Finland  FIN   
41  17   Paavo Johannes Aaltonen   M  28.0   175.0    64.0  Finland  FIN   
42  17   Paavo Johannes Aaltonen   M  28.0   175.0    64.0  Finland  FIN   
44  17   Paavo Johannes Aaltonen   M  28.0   175.0    64.0  Finland  FIN   

          games  year  season    city       sport  \
40  2014 Winter  2014  Winter   Sochi  Ice hockey   
41  1948 Summer  1948  Summer  London  Gymnastics   
42  1948 Summer  1948  Summer  London  Gymnastics   
44  1948 Summer  1948  Summer  London  Gymnastics   

                                     event   medal        bmi     ratio  \
40             Ice Hockey Men's Ice Hockey  Bronze  25.106333  2.164706   
41  Gymnastics Men's Individual All-Around  Bronze  20.897959  2.734375   
42        Gymnastics Men's Team All-Around    Gold  20.897959  2.734375   
44            Gymnastics Men's Horse Vault    Gold  20.897959  2.734375   

     bmi_category  
40     Overweight  
41  Normal weight  
42  Normal weight  
44  Normal weight

In [ ]:
ol['age'] = pd.to_numeric(ol['age'], errors='coerce')
ol['height'] = pd.to_numeric(ol['height'], errors='coerce')
ol['weight'] = pd.to_numeric(ol['weight'], errors='coerce')
ol['year'] = pd.to_numeric(ol['year'], errors='coerce')

In [ ]:
ol.head(4)

id                      name sex   age  height  weight     team  noc  \
40  16  Juhamatti Tapio Aaltonen   M  28.0   184.0    85.0  Finland  FIN   
41  17   Paavo Johannes Aaltonen   M  28.0   175.0    64.0  Finland  FIN   
42  17   Paavo Johannes Aaltonen   M  28.0   175.0    64.0  Finland  FIN   
44  17   Paavo Johannes Aaltonen   M  28.0   175.0    64.0  Finland  FIN   

          games  year  season    city       sport  \
40  2014 Winter  2014  Winter   Sochi  Ice hockey   
41  1948 Summer  1948  Summer  London  Gymnastics   
42  1948 Summer  1948  Summer  London  Gymnastics   
44  1948 Summer  1948  Summer  London  Gymnastics   

                                     event   medal        bmi     ratio  \
40             Ice Hockey Men's Ice Hockey  Bronze  25.106333  2.164706   
41  Gymnastics Men's Individual All-Around  Bronze  20.897959  2.734375   
42        Gymnastics Men's Team All-Around    Gold  20.897959  2.734375   
44            Gymnastics Men's Horse Vault    Gold  20.897959  2.734375   

     bmi_category  
40     Overweight  
41  Normal weight  
42  Normal weight  
44  Normal weight

#### `Changing to appropriate datatype`

In [ ]:
ol.info()

<class 'pandas.core.frame.DataFrame'>
Index: 30181 entries, 40 to 271103
Data columns (total 18 columns):
 #   Column        Non-Null Count  Dtype  
---  ------        --------------  -----  
 0   id            30181 non-null  int64  
 1   name          30181 non-null  object 
 2   sex           30181 non-null  object 
 3   age           30181 non-null  float64
 4   height        30181 non-null  float64
 5   weight        30181 non-null  float64
 6   team          30181 non-null  object 
 7   noc           30181 non-null  object 
 8   games         30181 non-null  object 
 9   year          30181 non-null  int64  
 10  season        30181 non-null  object 
 11  city          30181 non-null  object 
 12  sport         30181 non-null  object 
 13  event         30181 non-null  object 
 14  medal         30181 non-null  object 
 15  bmi           30181 non-null  float64
 16  ratio         30181 non-null  float64
 17  bmi_category  30181 non-null  object 
dtypes: float64(5), int64(2), obje

In [ ]:
ol['sex'] = ol['sex'].astype('category')
ol['team'] = ol['team'].astype('category')
ol['noc'] = ol['noc'].astype('category')
ol['games'] = ol['games'].astype('category')
ol['season'] = ol['season'].astype('category')
ol['sport'] = ol['sport'].astype('category')
ol['event'] = ol['event'].astype('category')
ol['medal'] = ol['medal'].astype('category')
ol['bmi_category'] = ol['bmi_category'].astype('category')

In [ ]:
ol.info()

<class 'pandas.core.frame.DataFrame'>
Index: 30181 entries, 40 to 271103
Data columns (total 18 columns):
 #   Column        Non-Null Count  Dtype   
---  ------        --------------  -----   
 0   id            30181 non-null  int64   
 1   name          30181 non-null  object  
 2   sex           30181 non-null  category
 3   age           30181 non-null  float64 
 4   height        30181 non-null  float64 
 5   weight        30181 non-null  float64 
 6   team          30181 non-null  category
 7   noc           30181 non-null  category
 8   games         30181 non-null  category
 9   year          30181 non-null  int64   
 10  season        30181 non-null  category
 11  city          30181 non-null  object  
 12  sport         30181 non-null  category
 13  event         30181 non-null  category
 14  medal         30181 non-null  category
 15  bmi           30181 non-null  float64 
 16  ratio         30181 non-null  float64 
 17  bmi_category  30181 non-null  category
dtypes: catego

#### `📌 Saving the cleaned datafame as a csv file: Olympics.csv`

In [ ]:
ol.to_csv('olympics.csv', index=False)

<h1 style="text-align: center;">END OF NOTEBOOK</h1>